ui/raidboss/data/05-shb/eureka/zadnor.js

```diff
@@ -81,30 +133,1155 @@ export default {
+      response: Responses.aoe(),
+    },
+    {
+      id: 'Zadnor Machines Fore-Hind Cannons',
+      netRegex: NetRegexes.startsUsing({ source: 'Kampe', id: '5CFF', capture: false }),
+      response: Responses.goSides(),
+    },
     // ***** Worn to a Shadow *****
+    {
+      id: 'Zadnor Shadow Bladed Beak',
+      // Not a cleave.
+      netRegex: NetRegexes.startsUsing({ source: 'Alkonost', id: '5E3B' }),
+      condition: tankBusterOnParty('shadow'),
+      response: Responses.tankBuster(),
+    },
+    {
+      id: 'Zadnor Shadow Nihility\'s Song',
+      netRegex: NetRegexes.startsUsing({ source: 'Alkonost', id: '5E3C', capture: false }),
+      condition: (data) => data.ce === 'shadow',
+      response: Responses.aoe(),
+    },
+    {
+      id: 'Zadnor Shadow Stormcall',
+      netRegex: NetRegexes.startsUsing({ source: 'Alkonost', id: '5E39', capture: false }),
+      condition: (data) => data.ce === 'shadow',
+      alertText: (_data, _matches, output) => output.text(),
+      outputStrings: {
+        text: {
+          en: 'Follow Slow Orb',
+        },
+      },
+    },
+    {
+      id: 'Zadnor Shadow Stormcall Away',
+      netRegex: NetRegexes.startsUsing({ source: 'Alkonost', id: '5E39', capture: false }),
+      condition: (data) => data.ce === 'shadow',
+      delaySeconds: 15,
+      infoText: (_data, _matches, output) => output.text(),
+      outputStrings: {
+        text: {
+          en: 'Away From Orb',
+        },
+      },
+    },
     // ***** A Familiar Face *****
+    {
+      id: 'Zadnor Face Ancient Quake IV',
+      netRegex: NetRegexes.startsUsing({ source: '4th-Make Hashmal', id: '5D14', capture: false }),
+      condition: (data) => data.ce === 'face',
+      response: Responses.aoe(),
+    },
+    {
+      id: 'Zadnor Face Rock Cutter',
+      netRegex: NetRegexes.startsUsing({ source: '4th-Make Hashmal', id: '5D13' }),
+      condition: tankBusterOnParty('face'),
+      response: Responses.tankBuster(),
+    },
+    {
+      id: 'Zadnor Face Extreme Edge Left',
+      netRegex: NetRegexes.startsUsing({ source: 'Phantom Hashmal', id: '5D0E', capture: false }),
+      alertText: (_data, _matches, output) => output.text(),
+      outputStrings: {
+        text: {
+          en: 'Find Phantom; Dodge Left',
+        },
+      },
+    },
+    {
+      id: 'Zadnor Face Extreme Edge Right',
+      netRegex: NetRegexes.startsUsing({ source: 'Phantom Hashmal', id: '5D0D', capture: false }),
+      alertText: (_data, _matches, output) => output.text(),
+      outputStrings: {
+        text: {
+          en: 'Find Phantom; Dodge Right',
+        },
+      },
+    },
+    {
+      id: 'Zadnor Face Hammer Round',
+      netRegex: NetRegexes.ability({ source: '4th-Make Hashmal', id: '5D10', capture: false }),
+      infoText: (_data, _matches, output) => output.text(),
+      outputStrings: {
+        text: {
+          en: 'Away From Hammer; Rotate Outside',
```

Comment:
Is the direction of the rotation implied?

Reply:
This is one of those "// TODO: figure out where the adds spawn and call clockwise or widdershins" sort of situations.  It's pretty obvious that you rotate away from the second hammer.